config/config-api/src/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -1308,26 +1309,39 @@ public PipelineConfig findPipelineUsingThisPipelineAsADependency(String pipeline
     }
 
     @Override
-    public Map<CaseInsensitiveString, List<CaseInsensitiveString>> templatesWithPipelinesForUser(String username, List<Role> roles) {
-        HashMap<CaseInsensitiveString, List<CaseInsensitiveString>> templateToPipelines = new HashMap<>();
-        for (PipelineTemplateConfig template : getTemplates()) {
-            if (isAuthorizedToAccessTemplate(new CaseInsensitiveString(username), roles, template)) {
-                templateToPipelines.put(template.name(), new ArrayList<>());
+    public List<TemplatesToPipelines> templatesWithPipelinesForUser(String username, List<Role> roles) {
```

Comment:
`username` is being passed along as string everywhere and then being converted to `CaseInsensitiveString` over here. But if you check all the callers, they already have the `CaseInsensitiveString` object for username, might as well pass that one. Given this is used by the service alone, and that the logic doesn't belong to the domain, would it make sense to move it to the service?